src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,236 @@
+   *
+   * @see Docs https://reach.tech/dialog#dialog-initialfocusref
+   */
+  initialFocusRef?: RefObject<HTMLDivElement>;
+  /**
+   * If the drawer is open
+   */
+  isOpen: boolean;
+  /**
+   * Which edge of the viewport should the drawer appear from
+   */
+  placement?: DrawerPlacementType;
+  /**
+   * Function that is called whenever the user either hits
+   *  the "Escape" key, clicks the close button icon, or clicks the overlay.
+   */
+  onDismiss?: (event?: React.SyntheticEvent) => void;
+  /**
+   * Title to be displayed at the top of the Drawer.
+   * A close button will be rendered automatically if this prop is defined.
+   */
+  title?: string;
+  /**
+   * The width of the Drawer when opened. Can be given a standard css value (px, rem, em, %),
+   * or a [width token](/?path=/story/design-tokens-design-tokens--page#width)
+   */
+  width?: DimensionSize | string;
+}
+
+export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
+  (
+    {
+      ariaLabel,
+      ariaLabelledBy,
+      allowPinchZoom = false,
+      children,
+      className,
+      closeButton = false,
+      closeOnOverlayClick = true,
+      containerRef = undefined,
+      dangerouslyBypassFocusLock = false,
+      dangerouslyBypassScrollLock = false,
+      hideOverlay = false,
+      initialFocusRef,
+      isOpen,
+      onDismiss,
+      placement = 'right',
+      title,
+      width = undefined,
+    },
+    ref,
+  ) => {
+    const dynamicWidth = (WIDTH_OPTIONS as Array<string | undefined>).includes(width)
+      ? `var(--size-width-${width})`
+      : width;
+
+    const dynamicStyle: CSSProperties = {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ['--w' as any]: dynamicWidth,
+    };
+
+    const overlayClassnames = classNames(styles.overlay, styles.drawer, {
+      [styles['hide-overlay']]: hideOverlay,
+      [styles['hide-overlay-left']]: hideOverlay && placement === 'left',
+      [styles['hide-overlay-right']]: hideOverlay && placement === 'right',
+      [styles['hide-overlay-bottom']]: hideOverlay && placement === 'bottom',
+      [styles['hide-overlay-top']]: hideOverlay && placement === 'top',
+      'position-fixed': containerRef === undefined,
+      'position-absolute': containerRef !== undefined,
+    });
+
+    const contentClassnames = classNames(styles['drawer-content'], {
+      [styles.bottom]: placement === 'bottom',
+      [styles.left]: placement === 'left',
+      [styles.right]: placement === 'right',
+      [styles.top]: placement === 'top',
+      [styles['hide-overlay']]: hideOverlay,
+      'overflow-auto': !closeButton && !title,
+      className,
+    });
+
+    const renderHeader = () => {
+      if (closeButton && onDismiss && !title) {
+        return (
+          <Box
+            alignItems="flex-end"
+            justifyContent="center"
+            padding="md lg"
+            borderWidth="0 0 xs 0"
+            className={styles['drawer-header']}
```

Comment:
Are we okay with splitting the styling across two places (Box props and CSS class)? My preference is to go with one or the other, but not both.

Reply:
Box props is preferred since they are typed, but the `drawer-header` class enables theming via component design tokens (aka css vars). It is the only thing that is "theme-able" for this particular part of the component. We've done this in other content container components like Modal, MediaModal, and Card

Reply:
If we update the type for `borderColor` to `borderColor?: BrandColor | `var(${string})`;`, then we could do
```suggestion
          <Box
            alignItems="flex-end"
            justifyContent="center"
            padding="md lg"
            borderWidth="0 0 xs 0"
            borderColor="var(--drawer-header-border-color, var(--color-brand-grey-100))"
```

Reply:
That would impact the `<Box>` and components using this pattern, so prefer to keep suggested change out of this scope and approach it holistically.